archive/c/c-plus-plus/reverse-string.cpp

```diff
@@ -0,0 +1,14 @@
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+int main(int argc, char *argv[])
+{
+    if(argc < 2)
+        return 0;
+    
+    std::string s = argv[1];
+    std::reverse(s.begin(), s.end());
+
+    std::cout << s << "\n";
```

Comment:
Maybe `std::endl` is a better choice here.

Reply:
Inserting a newline instead of `std::endl` is faster. You don't always want to flush the buffer.

Reply:
Plus, I've heard it's bad practice to always use `std::endl`.

Reply:
Well, I guess at least I can say I learned something new today. I genuinely thought `std::endl` was just set to `\n` on *nix and to `\r\n` on broken systems.

In that case, I guess we'll keep the `\n`